File: benchmarks/compare/string_passing.c
#include <stdio.h>
const char *pick(long long i) {
    if (i % 3 == 0) return "alpha";
    if (i % 3 == 1) return "beta";
    return "gamma";
}
const char *pass_through(const char *s) { return s; }
long long compute(long long n) {
    long long count = 0;
    long long i = 0;
    while (i < n) {
        const char *s = pass_through(pass_through(pick(i)));
        (void)s;
        if (i % 3 == 0) count++;
        i++;
    }
    return count;
}
int main(void) {
    printf("%lld\n", compute(15000000));
    return 0;
}

File: benchmarks/compare/string_passing.py
def pick(i):
    if i % 3 == 0:
        return "alpha"
    if i % 3 == 1:
        return "beta"
    return "gamma"

def pass_through(s):
    return s

def compute(n):
    count = 0
    i = 0
    while i < n:
        s = pass_through(pass_through(pick(i)))
        if i % 3 == 0:
            count = count + 1
        i = i + 1
    return count

print(compute(15000000))
